### ISSUE TITLE
New install erroring out on initial db upgrade for blog_file table

### ISSUE THREAD

**Issue by profmikegreene** [tsugiproject/tsugi#50]
https://github.com/tsugiproject/tsugi/blob/46ff55a61d3f18d5b49bf6dc06ae6f2cbb7d758f/admin/blob/database.php#L28

Is this a bug? Should it be `path(2048)`?

My admin/upgrade output below
```
Checking plugins table...
Checking for any needed upgrades...
Checking admin/lti/database.php ...
-- Checking table lti_key
-- Checking table lti_user
-- Checking table lti_context
-- Checking table lti_link
-- Checking table lti_link_activity
-- Checking table lti_membership
-- Checking table lti_event
-- Checking table lti_link_user_activity
-- Checking table lti_service
-- Checking table lti_result
-- Checking table lti_nonce
-- Checking table lti_domain
-- Checking table tsugi_string
-- Checking table profile
-- Checking table cal_event
-- Checking table cal_key
-- Checking table cal_context
-- Checking upgrade admin/lti/database.php 
-- Current data model version 201801271430 
Checking admin/key/database.php ...
-- Checking table key_request
-- Checking upgrade admin/key/database.php 
-- Current data model version 2014042200 
Checking admin/blob/database.php ...
-- Checking table blob_file
-- Creating table blob_file
Unable to create create table blob_file ( file_id INTEGER NOT NULL KEY AUTO_INCREMENT, file_sha256 CHAR(64) NOT NULL, context_id INTEGER NULL, link_id INTEGER NULL, file_name VARCHAR(2048), deleted TINYINT(1), contenttype VARCHAR(256) NULL, path VARCHAR(2048) NULL, content LONGBLOB NULL, blob_id INTEGER, json TEXT NULL, created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP, accessed_at TIMESTAMP NOT NULL DEFAULT '1970-01-02 00:00:00', INDEX `blob_indx_1` USING HASH (`file_sha256`), INDEX `blob_indx_2` (`path(256)`), CONSTRAINT `blob_ibfk_1` FOREIGN KEY (`context_id`) REFERENCES `lti_context` (`context_id`) ON DELETE SET NULL ON UPDATE CASCADE ) ENGINE = InnoDB DEFAULT CHARSET=utf8 42000:42000:SQLSTATE[42000]: Syntax error or access violation: 1072 Key column 'path(256)' doesn't exist in table
create table blob_file ( file_id INTEGER NOT NULL KEY AUTO_INCREMENT, file_sha256 CHAR(64) NOT NULL, context_id INTEGER NULL, link_id INTEGER NULL, file_name VARCHAR(2048), deleted TINYINT(1), contenttype VARCHAR(256) NULL, path VARCHAR(2048) NULL, content LONGBLOB NULL, blob_id INTEGER, json TEXT NULL, created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP, accessed_at TIMESTAMP NOT NULL DEFAULT '1970-01-02 00:00:00', INDEX `blob_indx_1` USING HASH (`file_sha256`), INDEX `blob_indx_2` (`path(256)`), CONSTRAINT `blob_ibfk_1` FOREIGN KEY (`context_id`) REFERENCES `lti_context` (`context_id`) ON DELETE SET NULL ON UPDATE CASCADE ) ENGINE = InnoDB DEFAULT CHARSET=utf8
```

**Comment by profmikegreene** [comment]
Changing that line to `path(2048)` didn't resolve my issue. Still debugging.

**Comment by profmikegreene** [comment]
Fixed by commit cb6c2da601a8f3695b8a9321704f48d5c7da4e11
